### PR TITLE
Sync with upstream kubeflow v1.7.0

### PR DESCRIPTION
**Pull request by mlassak**
Updates v1.6-branch to contain changes brought by upstream Kubeflow v1.7.0

## Description
Rebased opendatahub-io/kubeblow master branch onto upstream kubeflow/kubeflow master v1.7.0 tag, then merged to opendatahub-io/kubeflow v1.6-branch

## Merge criteria:
- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by maroroman** [comment]
@harshad16 @VaishnaviHire I tested this PR, Unit tests and E2E tests for kubeflow notebook controller and odh notebook controller worked fine. I did have to create a custom image, and during that I had to change the makefile of the kubeflow notebook controller somewhat because otherwise the controller-gen package would not install for me. I am assuming this is a local problem so I do not think changing the makefile is necessary.

As far as my testing went, I think the PR should be ready for merging

/lgtm

File: components/notebook-controller/config/manager/manager.yaml [hunk 1]
```diff
@@ -32,7 +32,27 @@ spec:
               configMapKeyRef:
                 name: config
                 key: ISTIO_GATEWAY
-        imagePullPolicy: Always
+          - name: CLUSTER_DOMAIN
+            valueFrom:
+              configMapKeyRef:
```
Comment (VaishnaviHire) [inline components/notebook-controller/config/manager/manager.yaml:37]: Configmap ref is added to the deployment. We should make sure, it matches with the configmap created by[ odh-dashboard](https://github.com/opendatahub-io/odh-dashboard/blob/main/backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts#L73).

File: components/notebook-controller/config/manager/manager.yaml [hunk 2]
```diff
@@ -32,7 +32,27 @@ spec:
...
+                name: config
+                key: CULL_IDLE_TIME
+          - name: IDLENESS_CHECK_PERIOD
+            valueFrom:
+              configMapKeyRef:
+                name: config
+                key: IDLENESS_CHECK_PERIOD
+        imagePullPolicy: IfNotPresent
```
Comment (VaishnaviHire) [inline components/notebook-controller/config/manager/manager.yaml:55]: imagePullPolicy should be updated to `Always`

File: components/notebook-controller/controllers/culling_controller.go [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+
+const (
+	KERNEL_EXECUTION_STATE_IDLE     = "idle"
+	KERNEL_EXECUTION_STATE_BUSY     = "busy"
+	KERNEL_EXECUTION_STATE_STARTING = "starting"
+)
+
+// Each kernel of the Notebook Server has a status.
+// KernelStatus struct:
+type KernelStatus struct {
```
Comment (VaishnaviHire) [inline components/notebook-controller/controllers/culling_controller.go:62]: I think we would also need to add `TerminalStatus` if it does not track terminal activity already

**Comment by harshad16** [comment]
Rechecked the commits. 
All of the commits from the v1.7 branch of Kubeflow seems to have been pulled in for branch 1.7


there are a few duplicate commits at the end, seems like they are due to merge conflicts.
Screenshot from 2023-05-25 15-44-17

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

thanks for the excellent work :100: :medal_sports:

**Comment by harshad16** [comment]
Lets merge this PR and fix the rest in a new subsequent PR.

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**